src/controllers/sourceController.ts

```diff
@@ -17,5 +17,30 @@ const router = express.Router();
 //Endpoints
 router.get('/api/sources', getAllSources);
 
+router.patch('/api/sources/:id', async (req, res) => {
+
+    // Get source ID from the URL parameter
+    const { id } = req.params; 
+
+    // Get the 'active' field from the request body
+    const { active } = req.body; 
+
```

Comment:
Add validation for the id if it is passed and if it is a number.
Add validation for the active property. Is it passed and is it a bool value.